tests/unit_tests/plugin_registries/test_plugin_registries.py

```diff
@@ -10,27 +10,217 @@
 # # # for more details. If you did not receive the license, for more information see:
 # # # https://github.com/U-S-NRL-Marine-Meteorology-Division/
 
-"""Test PluginRegistry Class."""
+"""TestPluginRegistry Class used for Unit Testing the Plugin Registries."""
 
-from geoips.plugin_registry import PluginRegistry
-import logging
 from glob import glob
-import pytest
+from importlib import import_module
+import logging
 from os.path import basename, splitext
+from pprint import pformat
+import pytest
+import json
 import yaml
 
+from geoips.plugin_registry import PluginRegistry, get_entry_point_group
+from geoips.errors import PluginRegistryError
+
 LOG = logging.getLogger(__name__)
 
 
-class TestPluginRegistry:
+class PluginRegistryValidator(PluginRegistry):
     """Subclass of PluginRegistry which adds functionality for unit testing."""
 
-    fpaths = glob(
+    def __init__(self, fpaths=None):
+        """Initialize TestPluginRegistry Class."""
+        super().__init__(fpaths)
+
+    def validate_plugin_types_exist(self, reg_dict, reg_path):
```

Comment:
So from validate_plugin_types through validate_registry_interfaces, this is all a direct copy/paste from geoips/plugin_registry.py with no additional udpates?

These methods were added to a new "PluginRegistryValidator" class in the test_plugin_registries.py test module (with an init added)?

Reply:
Yes, this is right. We discussed leaving them where they were and extracting the pytest portions, but determined that these tests really are only needed at development time and aren't needed at run time so it made sense to move them to the unit tests.